File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0, pro = 0;
        for(int i = 0; i<prices.size(); i++)
            for(int j = i+1; j<prices.size(); j++)
             {
                pro = prices[j] - prices[i];
                ans = max(ans, pro);
             }
        
        return ans;
         
    }
};

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        self.ans = float('-inf')
        def diam(root):
            if not root:
                return 0
            
            lh = diam(root.left)
            rh = diam(root.right)

            temp = lh + rh
            self.ans = max(temp, self.ans)

            return 1 + max(lh, rh)
        
        diam(root)
        return self.ans
